### PR TITLE
CFSet: add a test suite for creation, lookup and mutation

### PR DESCRIPTION
**Pull request by DTW-Thalion**
`CFSet.c` had only partial test coverage. This adds a self-contained CFSet test suite (`Tests/CFSet`) built on the existing `CFTesting.h` framework.

**Immutable (`create.m`):**
- `CFSetCreate`, `CFSetGetCount`.
- Lookup — `CFSetContainsValue`, `CFSetGetCountOfValue`, `CFSetGetValue`, `CFSetGetValueIfPresent`, `CFSetGetValues`.
- `CFSetApplyFunction` (verifies every value is visited once).
- `CFSetCreateCopy`, `CFSetGetTypeID` / `CFGetTypeID`, and `CFEqual` / `CFHash`.

**Mutable (`mutable.m`):**
- `CFSetCreateMutable`, `CFSetAddValue` (including that a value already present is not duplicated), `CFSetSetValue`, `CFSetReplaceValue` (present and absent), `CFSetRemoveValue`, `CFSetRemoveAllValues`, `CFSetCreateMutableCopy`.

This raises `CFSet.c` line coverage from 40.0% to 89.5% (measured with llvm-cov). All new tests pass and no existing tests are affected. The remaining uncovered lines are the toll-free-bridge (`CF_IS_OBJC`) paths and the internal formatting-description helper.

### CODE REVIEW

File: Tests/CFSet/create.m [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  otherValues[0] = CFSTR("a");
+  otherValues[1] = CFSTR("x");
+  other = CFSetCreate (NULL, otherValues, 2, &kCFTypeSetCallBacks);
+  PASS_CFNEQ(other, set, "Sets with different members are not equal.");
+  CFRelease (other);
+
+  CFRelease (set);
+
+  return 0;
+}
```
Comment (HendrikHuebner) [inline Tests/CFSet/create.m:98]: Maybe also add a brief test for empty sets.

File: Tests/CFSet/create.m [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  const void *out[3];
+  const void *found;
+  const void *otherValues[2];
+  CFIndex applied;
+  CFIndex i;
+  Boolean present;
+
+  values[0] = CFSTR("a");
+  values[1] = CFSTR("b");
+  values[2] = CFSTR("c");
```
Comment (HendrikHuebner) [inline Tests/CFSet/create.m:27]: Can we also have a test for sets created from an array of values with duplicates / NULLs?

File: Tests/CFSet/mutable.m [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  PASS_CF(CFSetGetCount (set) == 3 && CFSetContainsValue (set, CFSTR("a")),
+    "CFSetReplaceValue keeps a present value.");
+  CFSetReplaceValue (set, CFSTR("z"));
+  PASS_CF(CFSetGetCount (set) == 3 && !CFSetContainsValue (set, CFSTR("z")),
+    "CFSetReplaceValue ignores an absent value.");
+
+  copy = CFSetCreateMutableCopy (NULL, 0, set);
+  PASS_CF(copy != NULL, "CFSetCreateMutableCopy returns a set.");
+  PASS_CFEQ(copy, set, "A mutable copy is equal to the original.");
+  CFRelease (copy);
```
Comment (HendrikHuebner) [inline Tests/CFSet/mutable.m:41]: Also test whether removing an absent value is a no op.

**Review by HendrikHuebner — COMMENTED** [review]
Is the implementation of the mutable set generally the same as the immutable set for read only functions?

File: Tests/CFSet/create.m (hunk 1 above)
Comment (DTW-Thalion) [reply Tests/CFSet/create.m:98]: Done.

File: Tests/CFSet/mutable.m (hunk 3 above)
Comment (DTW-Thalion) [reply Tests/CFSet/mutable.m:41]: Done.

File: Tests/CFSet/create.m (hunk 2 above)
Comment (DTW-Thalion) [reply Tests/CFSet/create.m:27]: Adding that test surfaced a bug, so I've left it out of this suite rather than assert the broken behaviour: `CFSetCreate` doesn't de-duplicate. From `{a, b, a}` it returns count 3, and `CFSetGetValues` then hands back a garbage third pointer (segfault). Root cause is `GSHashTableCreate` incrementing `_count` per array element with no existing-key check, so `CFDictionaryCreate`/`CFBagCreate` are affected too. I'll fix that in a separate PR and add the de-dup test (and the NULL-callbacks case) there.

**Comment by DTW-Thalion** [comment]
@HendrikHuebner Yes — immutable and mutable CFSets are both backed by the same GSHashTable, and the read-only functions (`CFSetGetCount`, `CFSetContainsValue`, `CFSetGetValue`, `CFSetGetValues`, `CFSetGetCountOfValue`, `CFSetGetValueIfPresent`) just cast to `GSHashTableRef` and call the generic `GSHashTable*` accessors with no mutable/immutable branching, so reads behave identically for both.

**Comment by stefanbidi** [comment]
I'll approve these new tests because a bug report can always be added if it does not pass on macOS. At some point, we'll need someone to run these on our reference platform.
